receiver/prometheusreceiver/config.go

```diff
@@ -47,6 +47,9 @@ type Config struct {
 	// server in agent mode. This allows the user to call the endpoint to get
 	// the config, service discovery, and targets for debugging purposes.
 	APIServer APIServer `mapstructure:"api_server"`
+
+	// From feature gate.
+	enableNativeHistograms bool
```

Comment:
hmmmm.... I seem to recall there was a collector-wide effort to ensure all configs could be marshalled and unmarshalled to json without changes.  Not sure if we can add this here.

Reply:
It's not a config option , there is no need to marshal or unmarshal that field.
Lowercase fields are not marshaled or unmarshaled to json.
https://go.dev/play/p/jlD5KGokvpl

Also it gets really ugly if I add this elsewhere :(

Reply:
Well... It passes presubmits, so i'm assuming it is good to go